### ISSUE TITLE
Tune decoding hyperparams

### ISSUE THREAD

**Issue by KhoomeiK** [KhoomeiK/deredact#5]
`top_k_to_decode`: selects how many of the top logits to sample for checking their token's character count.
`min_p_logits`: once tokens that have the correct character count are found, this sets the minimum post-softmax probability for a token to be included for further decoding.
`prob_gamma`: up-weights the `min_p_logits` as the decoded sequence gets longer since possibilities naturally narrow as completions get longer.

Current hyperparam exploration has been in the ballpark of these:
```python
top_k_to_decode = 200
min_p_logits = 0.004
prob_gamma = 1.3
```

We may want to change the 2-stage filtering approach and deprecate the `top_k_to_decode` vs `min_p_logits` distinction.
We may want to set up a more sophisticated scheduler or weighting scheme for `min_p_logits`.

Issue #3 may introduce some new hyperparams as well related to batch sampling
